plugin/storage/badger/spanstore/reader.go

```diff
@@ -565,6 +565,10 @@ func (r *TraceReader) scanIndexKeys(indexKeyValue []byte, plan *executionPlan) (
 	return indexResults, err
 }
 
+func (s *TraceReader) Close() error {
```

Comment:
Why is it necessary? The reader/writer interfaces do not require Close.

Reply:
i was getting this error 
```bash
panic: interface conversion: *spanstore.TraceReader is not io.Closer: missing method Close [recovered]
        panic: interface conversion: *spanstore.TraceReader is not io.Closer: missing method Close
```
when on calling `e2eCleanUp` in the test, that is why i add added `Close` method
 ```
func (s *E2EStorageIntegration) e2eCleanUp(t *testing.T) {
	require.NoError(t, s.SpanReader.(io.Closer).Close())
	require.NoError(t, s.SpanWriter.(io.Closer).Close())
}
```

Reply:
I suggest we change that function to be conditional, e.g.
```
if closer, ok := s.SpanReader.(io.Closer); ok {
  require.NoError(t, closer.Close())
}
```

Reply:
i think we don't need to use it because now we are using SpanWriter and SpanReader from e2eInitialize(), which implements `Close` method